core/xchain/entitlement/evaluator.go

```diff
@@ -7,14 +7,21 @@ import (
 	"go.opentelemetry.io/otel/trace"
 
 	"github.com/towns-protocol/towns/core/config"
+	"github.com/towns-protocol/towns/core/contracts/base"
 	"github.com/towns-protocol/towns/core/node/crypto"
 	"github.com/towns-protocol/towns/core/node/infra"
+	"github.com/towns-protocol/towns/core/node/logging"
 )
 
 type Evaluator struct {
 	clients        BlockchainClientPool
 	evalHistrogram *prometheus.HistogramVec
-	ethChainIds    []uint64
+	// etherBasedChainIds includes any chain that uses ethereum as a native currency.
+	etherBasedChainIds []uint64
+	// ethereumChainIds refers to the list of actual ethereum mainnet and testnets
+	// this network is configured to support.
+	ethereumChainIds []uint64
```

Comment:
can you add a comment indicating that ethereumChainIds is a subset of etherBasedChainIds (or at least should be)